MixItUp.Base/Commands/CommandBase.cs

```diff
@@ -152,22 +157,32 @@ public Task Perform(UserViewModel user, IEnumerable<string> arguments = null)
 
         protected virtual async Task PerformInternal(UserViewModel user, IEnumerable<string> arguments, CancellationToken token)
         {
-            for (int i = 0; i < this.Actions.Count; i++)
+            if (this.IsRandomized)
             {
+                int selectedAction = this.random.Next(this.Actions.Count);
                 token.ThrowIfCancellationRequested();
 
-                if (this.Actions[i] is OverlayAction && ChannelSession.Services.OverlayServer != null)
+                await this.Actions[selectedAction].Perform(user, arguments);
```

Comment:
I'm not entirely sure this is going to be the best way.  Does the fact that we "don't do batching" matter? Are there other side effects?

Reply:
Also, there are some actions that "don't play nice". Like  a "Wait Action".  Would basically do nothing.  I just don't think it's worth adding smarts or anything, since, this would generally be used for small sets of random choices.

Reply:
Since it's in an Action Group, I'm fine with this logic. It actually works really nice! People would just make an Action Group of the choices themselves then just make a call to do one of them. For this sort of scenario, I think it works perfectly.

As for batching, we would want to still make sure we're doing that, in case the action selected in here is an Overlay action, either before, after, or during.